Gruntfile.js

```diff
@@ -38,7 +38,7 @@ module.exports = (grunt) => {
         library: libraryName,
         libraryTarget: 'umd',
         libraryExport: 'default',
-        globalObject: 'this',
+        globalObject: `typeof window !== 'undefined' ? window : this`,
```

Comment:
Why is this line necessary? In my PR #1072 I got the two examples to work without having to modify the gruntfile. 

From the webpack docs: "To make UMD build available on both browsers and Node.js, set output.globalObject option to 'this'."

https://webpack.js.org/configuration/output/#outputglobalobject

Reply:
It is wired, I have checked the history of `vexflow-debug.js` and found that 23c6ac0bbba1f907f0f7f3377a21147da1a0017f release as `typeof self !== 'undefined' ? self : this` while 0793189194ca216207b4ac1d2503156da24ebde7 revert to `window`.

It seems that a bug arisen between upgrading Webpack, you can see https://github.com/webpack/webpack/issues/6522.

Reply:
So currently I can revert this commit and re-release the folder to solve the problem.

Reply:
Hello! Do you have a small example which shows that this gruntfile change is necessary? An example which fails with the current gruntfile, and works better when we use your solution:

```
globalObject: `typeof window !== 'undefined' ? window : this`,
```

I confess that I am not a webpack and UMD expert. I am just going by the current webpack documentation: https://webpack.js.org/configuration/output/#outputglobalobject

The comment you reference is 3 years old. Could this have been solved by webpack in the meantime?

The two Node JS demos we currently have work fine without the gruntfile change. I'd like to see an example where it fails.

Reply:
You can reproduce like this:

```html
<script type="module" src="releases/vexflow-debug.js"></script>
```

```js
// releases/vexflow-debug.js
console.log(this); // => undefined
console.log(typeof window !== 'undefined' ? window : this); // => Window
```

Reply:
@ronyeh If you dig into the implementation of jQuery, you can find that they have adopted the style when exporting UMD style modules.

- REF: https://github.com/jquery/jquery/blob/a684e6ba836f7c553968d7d026ed7941e1a612d8/src/wrapper.js#L37

Reply:
Hi @aleen42 -- this change looks good overall. This is the only part I'm still confused about. I guess the question is: what breaks when you leave it as `this`? It seems like the demos work fine that way.

My suggestion would be to revert this line to keep it simple (unless, ofcourse, we can find a good reason not to.) :-)

Reply:
I think I now understand. Let me repro the issue and put together an example.

Reply:
I've tested the above on my branch. It does indeed fix a error with script type=module. Can we actually modify this line to:

```
globalObject: `typeof window !== 'undefined' ? window : typeof globalThis !== 'undefined' ? globalThis : this`,
```

This would also fix situations where both window and this are unavailable. Thanks!